mrmustard/lab_dev/states/states.py

```diff
@@ -115,7 +115,7 @@ class Number(Ket):
     """
 
     def __init__(
-        self, modes: Iterable[int], n: Union[int, Iterable[int]], cutoff: Optional[int] = None
+        self, modes: Sequence[int], n: Union[int, Sequence[int]], cutoff: Optional[int] = None
```

Comment:
DM and Ket have `modes = tuple[int, ...]`, wires has sets, here there are sequences. Choose one?

Reply:
Let's use tuples throughout (small, immutable). Only Wires will have sets. I'm not going to chase after all the type annotations in this PR though.